lib/bike_brigade_web/live/campaign_live/messaging_form_component.html.heex

```diff
@@ -53,10 +56,13 @@
     <div class="w-1/2">
       <%= label f, :details, class: "block text-sm font-medium leading-5 text-gray-700" do %>
         App Details
+        <C.button id="emoji-button" phx-hook="EmojiButtonHook" data-input-id={input_id(f, :details_body)} type="button" color={:white} size={:xxsmall}>
```

Comment:
I think you're missing what `input_id(f, :details_body)` does. 

This example from the docs may help https://hexdocs.pm/phoenix_html/Phoenix.HTML.Form.html#textarea/3

```
# Assuming form contains a User schema
textarea(form, :description)
#=> <textarea id="user_description" name="user[description]"></textarea>
```

Note that in this example we called `textarea` *without* and id, and it automatically got the id `user_description`

In our case we called the `textarea` function *with* an id parameter which overrides the default

```
<%= textarea instructions_template_form, :body, required: true,
            id: "text-body",
```

This means the resulting element was
```
<textarea id="text-body">
....
```

The `input_id` function https://hexdocs.pm/phoenix_html/Phoenix.HTML.Form.html#input_id/2 gives you the default id for an input field given it's name.

As I said in my previous comment, you could have written the button as
```
<C.button id="emoji-button" phx-hook="EmojiButtonHook" data-input-id="text-body" type="button" color={:white} size={:xxsmall}>
```

But we're using the `input_id` function which generates the id for us. 

Try getting rid of the id parameter entirely and seeing what the results are like in the inspector. See if you can get the code to work without hardcoding theid.